parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -75,8 +78,26 @@ protected ParquetValueReader<T> createReader(
     }
   }
 
-  protected abstract ParquetValueReader<T> createStructReader(
-      List<ParquetValueReader<?>> fieldReaders, Types.StructType structType);
+  /**
+   * This method can be overridden to provide a custom implementation which also uses the fieldId of
+   * the Schema when creating the struct reader
+   */
+  protected ParquetValueReader<T> createStructReader(
+      List<ParquetValueReader<?>> fieldReaders, Types.StructType structType, Integer fieldId) {
+    // Fallback to the signature without fieldId if not overridden
+    return createStructReader(fieldReaders, structType);
+  }
+
+  /**
+   * @deprecated will be removed in 1.11.0. Subclasses should override {@link
+   *     #createStructReader(List, Types.StructType, Integer)} instead
+   */
+  @Deprecated
+  protected ParquetValueReader<T> createStructReader(
```

Comment:
@pvary I made this Deprecated, no longer abstract, and removed it from the child classes GenericParquetReaders and InternalReader. 

This should mean we are still source compatible if someone was extending it but we no longer have any usages in library code

Reply:
With release of 1.10.0, should this method be removed in 1.12.0?